src/viam/sdk/common/exception.hpp

```diff
@@ -0,0 +1,52 @@
+/// @file common/exception.hpp
+///
+/// @brief Defines custom exceptions for the SDK.
+#pragma once
+#include <stdexcept>
+#include <string>
+
+#include <viam/sdk/resource/resource_api.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup Exception Classes related to SDK exceptions.
+
+/// @class ViamErrorCode
+/// @brief Defines a set of a error codes to be used in conjunction with
+/// ViamException.
+/// @ingroup Exception
+enum class ViamErrorCode {
```

Comment:
You probably don't need `Viam` on the name.

Reply:
Removed.